### PR TITLE
feat: common helper function for selecting linked accounts

### PR DESCRIPTION
**Pull request by danielgent**
Lol!  I fixed existing bug on `main` on https://github.com/sesamyab/auth/pull/316

*when that is merged* rebase this PR onto `main` and should be all good :muscle:

### CODE REVIEW

File: integration-test/management-api/users-by-email.spec.ts [hunk 1]
```diff
@@ -265,6 +265,10 @@ describe("users by email", () => {
         provider: "email",
         user_id: barEmailId.split("|")[1],
         isSocial: false,
+        profileData: {
+          email: "bar@example.com",
+          email_verified: false,
+        },
```
Comment (danielgent) [inline integration-test/management-api/users-by-email.spec.ts:271]: I'm now using the new `enrichUser` that merges in `profileData`

File: src/routes/management-api/users.ts [hunk 2]
```diff
@@ -125,52 +126,9 @@ export class UsersMgmtController extends Controller {
...
-    );
-
-    const { id, ...userWithoutId } = user;
-
-    return {
-      ...userWithoutId,
-      identities: [userIdentity, ...linkedProfileIdentities],
-      user_id: user.id,
-    };
+    return userResponse;
```
Comment (danielgent) [inline src/routes/management-api/users.ts:131]: no change here (after fixing the bug here on https://github.com/sesamyab/auth/pull/316 :smile: )

File: src/routes/management-api/users.ts [hunk 3]
```diff
@@ -296,10 +249,6 @@ export class UsersMgmtController extends Controller {
       q: `linked_to:${body.link_with}`,
     });
 
-    // we're doing this mapping very frequently... once we include profileData
-    // would make sense to have a util function. TBD
-    // should this be type UserIdentities? e.g. first one does not have profileData, the rest do
-    // check Auth0 and actually link an account
```
Comment (danielgent) [inline src/routes/management-api/users.ts:302]: I nuked all these comments. I've made tickets to discuss what we do with the other mgmt API routes going forward

Not the most urgent thing to do

File: src/utils/enrichUser.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { Env, User } from "../types";
+import { UserResponse } from "../types/auth0/UserResponse";
+import { Identity } from "../types/auth0/Identity";
+
+export async function enrichUser(
+  env: Env,
+  tenantId: string,
+  primaryUser: User,
+): Promise<UserResponse> {
+  const linkedusers = await env.data.users.list(tenantId, {
```
Comment (danielgent) [inline src/utils/enrichUser.ts:11]: :see_no_evil: I'm aware that the double SQL call, on edge, will mean a large roundtrip time if the user is in China (such is how Edge works!)

*but for now* I'm happy with this just to get it built for users in Europe :smiling_face_with_tear: 

I figure if we get the tests running with SQLite then we can move more logic into the SQL queries *and have them be tested* :partying_face:

Comment (markusahlstrand) [reply src/utils/enrichUser.ts:11]: Think this will mostly be an issue when listing users are we need to make one query per user

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**